File: guanabara-java/src/com/danillo/algoritmos/lista05/Exercicio60.java
package com.danillo.algoritmos.lista05;

import java.util.Scanner;

public class Exercicio60 {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);

        String nomePessoaMaisVelha = null, nomeMulherMaisJovem = null;
        int homensMais30anos = 0, mulherMaisJovem = Integer.MAX_VALUE, maiorIdade = 0, soma = 0, quantidadePessoas = 0;
        char continuar = 'S';

        while(continuar == 'S'){
            System.out.println("Insira o nome: ");
            String nome = entrada.nextLine();
            System.out.println("Insira a idade: ");
            int idade = entrada.nextInt();
            System.out.println("Insira o gênero: ");
            char genero = entrada.next().charAt(0);
            
            if (idade > maiorIdade){
                maiorIdade = idade;
                nomePessoaMaisVelha = nome;
            }
            
            if (genero == 'F' && idade < mulherMaisJovem){
                mulherMaisJovem = idade;
                nomeMulherMaisJovem = nome;
            }

            if(genero == 'M' && idade > 30){
                homensMais30anos++;
            }

            soma = soma + idade;
            quantidadePessoas++;

            System.out.println("Deseja continuar? (S ou N)");
            continuar = entrada.next().charAt(0);
            entrada.nextLine();
        }

        float mediaIdade = (float)soma/quantidadePessoas;

        System.out.println("Nome da pessoa mais velha: " +nomePessoaMaisVelha+
                           "\nNome da mulher mais jovem: " +nomeMulherMaisJovem+
                           "\nMédia de idade do grupo: " +mediaIdade+
                           "\nQuantidade de homens com mais de 30 anos: " +homensMais30anos);
    }
}

File: guanabara-python/src/lista05/exercicio60.py
continuar = 1
homens_acima_30 = 0
mulheres_menores_18 = 0
maior_idade = 0
soma_idade = 0
i = 0

while continuar == 1:
    nome = input("Insira o nome: ")
    idade = int(input("Insira a idade: "))
    genero = int(input("Insira o gênero: 1 - Masculino 2 - Feminino"))

    if idade > maior_idade:
        nome_pessoa_mais_velha = nome

        if genero == 2:
            nome_mulher_mais_velha = nome

    soma_idade = soma_idade + idade

    if genero == 1 and idade > 30:
        homens_acima_30 = homens_acima_30 + 1

    if genero == 2 and idade < 18:
        mulheres_menores_18 = mulheres_menores_18 + 1

    i = i + 1
    continuar = int(input("Deseja continuar? 1 - Sim | 2 - Não"))

media_idade = soma_idade / i

print(f"Nome da pessoa mais velha: {nome_pessoa_mais_velha}.\nNome da mulher mais velha: {nome_mulher_mais_velha}."
      f"\nMédia de idade: {media_idade}.\nHomens com mais de 30 anos: {homens_acima_30}."
      f"\nMulheres com menos de 18 anos: {mulheres_menores_18}.")
